server/src/fixed-asset/dto/create-asset.dto.ts

```diff
@@ -0,0 +1,75 @@
+import { IAsset } from '../interfaces/asset.interface';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { IsNotEmpty, IsString, MinLength, MaxLength, Matches, IsOptional, IsNumber, IsPositive } from 'class-validator';
+import { Transform } from 'class-transformer';
+import { ASSET_NAME_REGEX, ASSET_DATE_REGEX } from '../../constants/regex.constants';
+import { VALIDATION_MESSAGES } from '../../constants/validation-messages';
+
+export class CreateAssetDto implements IAsset {
+  @ApiProperty({ required: true, example: 'Стіл' })
+  @IsNotEmpty()
+  @IsString()
+  @Transform(({ value }: { value: string }) => value.trim())
+  @MinLength(2, { message: VALIDATION_MESSAGES.ASSET.NAME_MIN_LENGTH })
+  @MaxLength(100, { message: VALIDATION_MESSAGES.ASSET.NAME_MAX_LENGTH })
+  @Matches(ASSET_NAME_REGEX, { message: VALIDATION_MESSAGES.ASSET.NAME_INVALID_CHARACTERS})
+  name: string;
+
+  @ApiPropertyOptional({ example: 'Склад 2'})
+  @IsOptional()
```

Comment:
Can I ask you to check if the validation would be passed without that annotation-marker when the value wasn't passed?